src/tools/fuzzing.h

```diff
@@ -507,11 +607,28 @@ class TranslateToFuzzReader {
   std::map<Type, std::vector<Index>>
     typeLocals; // type => list of locals with that type
 
+  void prepareToCreateFunctionContents(Function* func_) {
+    func = func_;
+    labelIndex = 0;
+    assert(breakableStack.empty());
+    assert(hangStack.empty());
+  }
+
+  void finishCreatingFunctionContents() {
+    if (HANG_LIMIT > 0) {
+      addHangLimitChecks(func);
+    }
+    typeLocals.clear();
```

Comment:
Clearing function-local state manually seems like a code smell. If we need to add more state in the future, it would be easy to forget to clear it. Could this be handled automatically via RAII with some sort of `FunctionCreationContext` object?

Reply:
Fair enough, but how about in a later PR? Most of this is pre-existing, and this PR is already doing a lot.

Reply:
Sure, sounds good.